docs/data/joy/components/autocomplete/autocomplete.md

```diff
@@ -0,0 +1,355 @@
+
+1. The value for the textbox must be chosen from a predefined set of allowed values, e.g., a location field must contain a valid location name: [combo box](#combo-box).
+2. The textbox may contain any arbitrary value, but it is advantageous to suggest possible values to the user, e.g., a search field may suggest similar or previous searches to save the user time: [free solo](#free-solo).
+
+It's meant to be an improved version of the "react-select" and "downshift" packages.
+
+{{"component": "modules/components/ComponentLinkHeader.js"}}
+
+## Combo box
+
+The `Autocomplete` renders an input and a popup, after triggered by a pointer or keyboard, that displays a list of items from the `options` prop. The value must be chosen from a predefined set of allowed values.
+
+{{"demo": "ComboBox.js"}}
+
+### Options structure
+
+By default, the component accepts the following options structures:
+
+```ts
+interface AutocompleteOption {
+  label: string;
+}
+// or
+type AutocompleteOption = string;
+```
+
+for instance:
+
+```js
+const options = [
+  { label: 'The Godfather', id: 1 },
+  { label: 'Pulp Fiction', id: 2 },
+];
+// or
+const options = ['The Godfather', 'Pulp Fiction'];
+```
+
+However, you can use different structures by providing a `getOptionLabel` prop:
+
+```js
+const options = [
+  { title: 'Pulp Fiction', id: 2 },
+  // ...
+];
+
+<Autocomplete getOptionLabel={option => option.title}>
+```
+
+### Controlled states
+
+The component has two states that can be controlled:
+
+1. the "value" state with the `value`/`onChange` props combination. This state represents the value selected by the user, for instance when pressing <kbd class="key">Enter</kbd>.
+2. the "input value" state with the `inputValue`/`onInputChange` props combination. This state represents the value displayed in the textbox.
+
+:::warning
+⚠️ These two states are isolated, they should be controlled independently.
+:::
+
+{{"demo": "ControllableStates.js"}}
+
+## Label
+
+Put an `Autocomplete`, a `FormLabel` and a `FormHelperText` (optional) under a `FormControl` component.
+
+{{"demo": "LabelAndHelperText.js"}}
+
+## Features
+
+Each of the following examples demonstrates one feature of the Autocomplete component.
+
+{{"demo": "Playground.js"}}
+
+## Free solo
+
+Set `freeSolo` to true so the textbox can contain any arbitrary value.
+
+### Search input
+
+The prop is designed to cover the primary use case of a **search input** with suggestions, e.g. Google search or react-autowhatever.
+
+{{"demo": "FreeSolo.js"}}
+
+:::warning
+⚠️ Be careful when using the free solo mode with non-string options, as it may cause type mismatch.
+
+The value created by typing into the textbox is always a string, regardless of the type of the options.
+:::
+
+### Creatable
+
+If you intend to use this mode for a [combo box](#combo-box) like experience (an enhanced version of a select element) we recommend setting:
```

Comment:
"for a [combo box](#combo-box) like experience" -> I do not understand what this means

Reply:
The explanation about combo box is explained in https://deploy-preview-34315--material-ui.netlify.app/joy-ui/react-autocomplete/#combo-box.

I think this is okay because users can click on the [combo box](#combo-box) and it will link to the explanation.

Reply:
Gottcha.